cloudinit/sources/DataSourceAzure.py

```diff
@@ -434,20 +450,30 @@ def crawl_metadata(self):
         # need to look in the datadir and consider that valid
         ddir = self.ds_cfg['data_dir']
 
+        # The order in which the candidates are inserted matters here, because
+        # it determines the value of ret. More specifically, the first one in
+        # the candidate list determines the path to take in order to get the
+        # metadata we need.
         candidates = [self.seed_dir]
         if os.path.isfile(REPROVISION_MARKER_FILE):
             candidates.insert(0, "IMDS")
+        elif os.path.isfile(REPROVISION_NIC_ATTACH_MARKER_FILE):
+            candidates.insert(0, "NIC_ATTACH_MARKER_PRESENT")
```

Comment:
While we are here, can you emit telemetry using `report_diagnostic_event` if either of these marker files exist?
This will allow enable ease of debugging when there is a reboot during the reprovision workflow.

```suggestion
        if os.path.isfile(REPROVISION_MARKER_FILE):
            candidates.insert(0, "IMDS")
            report_diagnostic_event(
                'Reprovision marker file already present before crawling Azure metadata: %s' % REPROVISION_MARKER_FILE,
                logger=LOG.debug)
        elif os.path.isfile(REPROVISION_NIC_ATTACH_MARKER_FILE):
            candidates.insert(0, "NIC_ATTACH_MARKER_PRESENT")
            report_diagnostic_event(
                'Reprovision nic attach marker file already present before crawling Azure metadata: %s' % REPROVISION_NIC_ATTACH_MARKER_FILE,
                logger=LOG.debug)
```